### PR TITLE
Fix behavior of RemoteData.getfile()

### PR DESCRIPTION
**Pull request by Crivella**
The :return: value in the documentation was not correct.

### CODE REVIEW

File: aiida/orm/nodes/data/remote.py [hunk 1]
```diff
@@ -57,9 +57,9 @@ def getfile(self, relpath, destpath):
         """
         Connects to the remote folder and gets a string with the (full) content of the file.
```
Comment (ltalirz) [inline aiida/orm/nodes/data/remote.py]:
```suggestion
        Connects to the remote folder and retrieves the content of a file.
```

File: aiida/orm/nodes/data/remote.py [hunk 2]
```diff
@@ -57,9 +57,9 @@ def getfile(self, relpath, destpath):
...
 
-        :param relpath: The relative path of the file to show.
-        :param destpath: A path on the local computer to get the file
-        :return: a string with the file content
+        :param relpath:  The relative path of the file on the remote to retrieve.
+        :param destpath: The absolute path of the copied/retrieved file on the local machine.
+        :return: a list containing the file names in 'destpath'
```
Comment (ltalirz) [inline aiida/orm/nodes/data/remote.py]: Indeed, this seems to be what it is doing... 
@giovannipizzi Is there a good reason for doing that (e.g. potentially returning the names of files that have nothing to do with the one that was just transferred?)

Comment (ltalirz) [reply aiida/orm/nodes/data/remote.py]: Just coming back to this - this is actually a bug, right @sphuber ?

This

**Comment by Crivella** [comment]
Actually as of know this method would have the double behaviour of copying the file to the local machine and returning its content as a string.

I'm not sure if it would be better to split this two into separate methods.

**Comment by giovannipizzi** [comment]
Hi all.
Indeed, these were utility functions I added at some point. I think the return value is a copy-paste error, so OK to not return anything.
So for this PR I would just remove the return value (and, ideally, add a test for this and for listdir).

More generally, let's discuss in #1857 if we want to keep these functions or how to make it clear that they open an transport and you do not want to do this in a workflow (see also #3478 )

**Comment by ltalirz** [comment]
@giovannipizzi 
Since I think we agree that the return value of this function was really a bug, I think it's fine to change without a deprecation warning.

Please review.

**Comment by giovannipizzi** [comment]
Thanks a lot @Crivella and @ltalirz!
@sphuber can I update this branch and merge? Or are you working on merging? (Ok I you want to take care of this.

**Comment by sphuber** [comment]
Thanks guys, I will merge another PR first that is already good to go and will then update this and merge it

**Comment by sphuber** [comment]
Ok, the other PR has problems with the tests, so I will merge this first after all

**Pull request merged**